src/sqlalchemy_cratedb/dialect.py

```diff
@@ -233,6 +233,30 @@ def connect(self, host=None, port=None, *args, **kwargs):
             return self.dbapi.connect(servers=servers, **kwargs)
         return self.dbapi.connect(**kwargs)
 
+    def do_execute(self, cursor, statement, parameters, context=None):
+        """
+        Slightly amended to store its response into the request context instance.
+        """
+        result = cursor.execute(statement, parameters)
+        if context is not None:
```

Comment:
Whe can this happen? (context is `None`)

Reply:
I can't tell when/why this can happen, but it might be due to backwards-compatibility reasons. This is the signature of the methods being overridden:

```python
def do_executemany(self, cursor, statement, parameters, context=None):
    cursor.executemany(statement, parameters)
```

See also https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_35/lib/sqlalchemy/engine/default.py#L937-L944.

Reply:
thx!